controller/src/main/java/org/jboss/as/controller/ModelControllerImpl.java

```diff
@@ -1259,6 +1220,7 @@ private void publish() {
             ModelControllerImpl.this.managementModel.set(this);
             ControllerLogger.MGMT_OP_LOGGER.tracef("published %s", this);
             published = true;
+            capabilityRegistry.publish();
```

Comment:
The ordering here seems wrong. It may not matter, but it seems unintuitive to log and set the flag, and then continue with the publishing process.